js/wasm/ethkey/src/main.rs

```diff
@@ -0,0 +1,153 @@
+// Copyright 2015-2017 Parity Technologies (UK) Ltd.
+// This file is part of Parity.
+
+// Parity is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+#![feature(lang_items, core_intrinsics)]
+#![feature(start)]
+#![feature(link_args)]
+#![no_std]
+use core::intrinsics;
+
+// Pull in the system libc library for what crt0.o likely requires.
+extern crate libc;
+extern crate tiny_keccak;
+extern crate tiny_secp256k1;
+
+use tiny_secp256k1::{is_valid_secret, create_public_key, ECPointG};
+
+#[link_args = "-s EXPORTED_FUNCTIONS=['_input_ptr','_secret_ptr','_public_ptr','_address_ptr','_ecpointg','_verify_secret','_brain']"]
```

Comment:
they should also be marked `extern` to force them to adhere to C calling convention. otherwise they are only safe to call from Rust code